src/main.rs

```diff
@@ -548,4 +549,47 @@ mod tests {
         let names: Vec<_> = group.iter_mut().map(|s| s.name.as_str()).collect();
         assert_eq!(vec!["boo", "bar"], names);
     }
+
+    #[test]
+    fn check_env_config_without_clear() {
+        set_var("GLOBAL_VAR_123", "present");
+
+        let yaml = "
+            name: bar
+            key: b
+            env:
+              FOO: bar
+            cmd: echo -n \"The value of FOO is $FOO and GLOBAL_VAR_123 is $GLOBAL_VAR_123\"
+        ";
```

Comment:
I think we test too much here. If YAML format will change, it will break this test. It's better to create tasks using the `Task` structure directly. It is a little more cumbersome and repetitive (in which case you can extract a helper method to create tasks), but more isolated and robust.